### DISCUSSION TITLE
New Referral System

### DISCUSSION THREAD

**Discussion by Dionis404** [sunflower-land/sunflower-land#3903] [Ideas]
**Registration Link:**
- The player creates a registration link containing their farm number.
- In the player's personal cabinet, there is an option to customize the referral link by adding bonus resources or gifts (e.g., resources, food, etc.) that they are willing to give to a new player.
- This helps new players get started and receive assistance from the friend who invited them.

**Registration Process and Initial Bonuses:**
- A new player who clicks on the link receives a welcome message and sees a list of gifts that are given to them upon registration.
- When registering, the new player automatically receives these bonuses on their account.

**Referrer Rewards:**
- A community coin is given out immediately after the new player completes the registration.
- 5 block bucks are given out after the new player makes their first purchase in the game.

**Additional Reward Tiers:**
- Introduce a reward level system for referrer rewards. For example, for every fifth invited active player, the referrer receives an additional bonus (e.g., a rare resource or a unique item).
- Achievement program: For referrers who have invited a certain number of active players, special achievements with unique rewards can be created.

**Tracking and Analytics:**
- In the referrer's personal cabinet, a statistics section is added where they can see how many players registered through their link, how many of them made their first purchase, and what rewards they received.

**Anti-Multi-account Measures:**
- Set limits on the amount and value of resources given to new players. 
- Require new players to reach a certain level or complete specific tasks before referrers receive rewards.

**Comment by TheDcol** [comment]
Hey mate! just a question:

> In the player's personal cabinet, there is an option to customize the referral link by adding bonus resources or gifts (e.g., resources, food, etc.) that they are willing to give to a new player.

Are the resources taken out of your farm when you create the link or when the new player creates their farm? If it's the former, where would these resources sit in the meantime?

My other comment was that there should be a Community Coin Shop established first

**Reply by Dionis404** [reply]
Resources are extracted from your farm immediately when you create a link. 
In this way we can secure the game from various abuses. 

When a player creates a referral link and adds resources to it, it would be logical if this link has a limit on the number of invitees. 

I think this system can help to attract new players, streamers and bloggers.

**Reply by Dionis404** [reply]
I have written my thoughts on the topic of community coin, you may find them interesting. 
https://github.com/sunflower-land/sunflower-land/discussions/3924
I would be glad to receive feedback.
